test/powershell/Modules/Microsoft.PowerShell.Management/Remove-Item.Tests.ps1

```diff
@@ -1,170 +1,195 @@
-	    # Create multiple files with specific string
-	    New-Item -Name file1.wav -Path $testpath -ItemType "file" -Value "lorem ipsum"
-	    New-Item -Name file2.wav -Path $testpath -ItemType "file" -Value "lorem ipsum"
+    Context "Directory Removal Tests" {
+        BeforeAll {
+            $testdirectory = Join-Path -Path $testpath -ChildPath testdir
+            $testsubdirectory = Join-Path -Path $testdirectory -ChildPath subd
+        }
 
-	    # Create a single file that does not match that string
-	    New-Item -Name file1.txt -Path $testpath -ItemType "file" -Value "lorem ipsum"
+        BeforeEach {
+            New-Item -Name "testdir" -Path $testpath -ItemType "directory" -Force
 
-	    # Delete the specific string
-	    Remove-Item (Join-Path -Path $testpath -ChildPath "file*") -Exclude *.wav -Include *.txt
+            Test-Path $testdirectory | Should -BeTrue
+        }
 
-	    # validate that the string under test was deleted, and the nonmatching strings still exist
-	    Test-Path (Join-Path -Path $testpath -ChildPath file1.wav) | Should -BeTrue
-	    Test-Path (Join-Path -Path $testpath -ChildPath file2.wav) | Should -BeTrue
-	    Test-Path (Join-Path -Path $testpath -ChildPath file1.txt) | Should -BeFalse
+        It "Should be able to remove a directory" {
+            { Remove-Item $testdirectory -ErrorAction Stop } | Should -Not -Throw
 
-	    # Delete the non-matching strings
-	    Remove-Item (Join-Path -Path $testpath -ChildPath file1.wav)
-	    Remove-Item (Join-Path -Path $testpath -ChildPath file2.wav)
+            Test-Path $testdirectory | Should -BeFalse
+        }
 
-	    Test-Path (Join-Path -Path $testpath -ChildPath file1.wav) | Should -BeFalse
-	    Test-Path (Join-Path -Path $testpath -ChildPath file2.wav) | Should -BeFalse
-	}
-    }
+        It "Should be able to recursively delete subfolders" {
+            New-Item -Name "subd" -Path $testdirectory -ItemType "directory"
+            New-Item -Name $testfile -Path $testsubdirectory -ItemType "file" -Value "lorem ipsum"
 
-    Context "Directory Removal Tests" {
-	$testdirectory = Join-Path -Path $testpath -ChildPath testdir
-	$testsubdirectory = Join-Path -Path $testdirectory -ChildPath subd
-	BeforeEach {
-	    New-Item -Name "testdir" -Path $testpath -ItemType "directory" -Force
+            $complexDirectory = Join-Path -Path $testsubdirectory -ChildPath $testfile
+            Test-Path $complexDirectory | Should -BeTrue
 
-	    Test-Path $testdirectory | Should -BeTrue
-	}
+            { Remove-Item $testdirectory -Recurse -ErrorAction Stop } | Should -Not -Throw
 
-	It "Should be able to remove a directory" {
-	    { Remove-Item $testdirectory } | Should -Not -Throw
+            Test-Path $testdirectory | Should -BeFalse
+        }
 
-	    Test-Path $testdirectory | Should -BeFalse
-	}
+        It "Should be able to recursively delete a directory with a trailing backslash" {
```

Comment:
Thanks for adding this regression test!

Reply:
Really we have _very_ small code coverage. I hope WGs start work and we add more tests to exclude regressions.